music/src/music_rnn/model.ts

```diff
@@ -275,17 +309,29 @@ export class MusicRNN {
       } else {
         const logits =
             lastOutput.matMul(this.lstmFcW).add(this.lstmFcB) as tf.Tensor2D;
-        const sampledOutput =
-            (temperature ?
-                 tf.multinomial(
-                       logits.div(tf.scalar(temperature)) as tf.Tensor2D, 1)
-                     .as1D() :
-                 logits.argMax(1).as1D());
+
+        let sampledOutput;
+        if (computeProbs) {
+          // We are also returning the sequence of probabilities, so calculate
+          // those before sampling.
+          const theseProbs = temperature ?
```

Comment:
For simplicity, you can just compute the probs in both cases and use it in the multinomial. The only time you shouldn't compute the probs is if computeProbs is false and temperature is not defined.